core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ArraySortAfterArrayDistinct.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.iterative.rule;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.metadata.Metadata;
+import io.trino.sql.planner.ArraySortAfterArrayDistinctRewriter;
+import io.trino.sql.planner.TypeAnalyzer;
+import io.trino.sql.planner.iterative.Rule;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class ArraySortAfterArrayDistinct
+        extends ExpressionRewriteRuleSet
+{
+    public ArraySortAfterArrayDistinct(Metadata metadata, TypeAnalyzer typeAnalyzer)
+    {
+        super(createRewrite(metadata, typeAnalyzer));
+    }
+
+    @Override
+    public Set<Rule<?>> rules()
```

Comment:
Why do you override this method?

Reply:
I based myself on another code that was having these rules setup. I will test without this Override and see if I can get rid of that.

Reply:
Actually this is needed, as following code can't handle aggregations as it can't resolve functions like count (window functions):
` ResolvedFunction decodeFunction(QualifiedName name);`